_posts/2023-02-13-Branch-Review-and-Learn.md

```diff
@@ -0,0 +1,135 @@
+      'git1': '#10B981',
+      'gitBranchLabel1': '#ffffff'
+} } }%%
+    gitGraph
+       commit id: "..."
+       commit id: "opened"
+       branch learner
+       checkout learner
+       commit id: "completed"
+       checkout main
+</pre>
+
+_An example git diagram showing learner branch based off main._
+{:.center}
+
+Following the course assignment workflow, the diagram above shows an in-progress `learner` branch based off of the `main` branch. When the `learner` branch is created, we bring into it everything we know from `main` (the course) so far in the form of [commits](https://github.com/git-guides/git-commit), or groups of changes to various files. [Branching](https://github.com/git-guides#create-a-branch) allows us to make consistent and well described changes based on what's already happened without impacting others work in the meantime.
+
+> Branching best practices:
+>
+> - __Keep the name and work with branches dedicated to a specific and focused purpose.__ For example: a branch named `fix-links-in-docs` might entail work related to fixing HTTP links within documentation.
+> - __Consider the use of [Github Forks](https://docs.github.com/en/get-started/quickstart/fork-a-repo)__ (along with branches within the fork) to help further isolate and enrich work potential. Forks also allow remixing existing work into new possibilities.
+> - __[_festina lente_](https://en.wikipedia.org/wiki/Festina_lente) or "make haste, slowly":__ Commits on any branch represent small chunks of a cohesive idea which will eventually be brought to `main`. It is often beneficial to be consistent with small, gradual commits to avoid a rushed or incomplete submission. The same applies more generally for software; taking time upfront to do things well can mean time saved later.
+
+## Reviewing the Branched Work
+
+<pre class="mermaid">
+%%{init: { 'logLevel': 'debug', 'theme': 'default' , 'themeVariables': {
+      'git0': '#6366F1',
+      'git1': '#10B981',
+      'gitBranchLabel1': '#ffffff'
+} } }%%
+    gitGraph
+       commit id: "..."
+       commit id: "opened"
+       branch learner
+       checkout learner
+       commit id: "completed"
+       checkout main
+       merge learner id: "reviewed"
+</pre>
+
+_An example git diagram showing learner branch being merged with main after a review._
+{:.center}
+
+The diagram above depicts a merge from the `learner` branch to pull the changes into the `main` branch, simulating an assignment being returned for review within a course. While merges may be forced without review, it's a best practice to ask others for a [Pull Request (PR) Review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) (also known as a [Merge Request (MR)](https://docs.gitlab.com/ee/user/project/merge_requests/) on some systems). Doing this provides a chance to make revisions before code changes are "finalized" within the `main` branch.
```

Comment:
I'm being more than a little pedantic here, but technically a developer issues a pull request, then the PR is reviewed (or not), and then potentially merged. IMHO it's important to identify that the pull request is its own entity, separate from the review. Maybe you could reword this as "While merges may be forced without review, it's a best practice create a Pull Request (PR) and then ask other members of your team to review it."

Reply:
Thanks, I like that change, I'll add that clarity into the wording here.